
diff -ru ushare-1.1a-orig/configure ushare-1.1a/configure
--- ushare-1.1a-orig/configure	2007-12-09 13:03:36.000000000 +0000
+++ ushare-1.1a/configure	2015-05-06 15:20:14.466687537 +0100
@@ -437,7 +437,6 @@
   *)
   echo "Unknown option \"$opt\"."
   echo "See $0 --help for available options."
-  exit 1
   ;;
   esac
 done
diff -ru ushare-1.1a-orig/src/Makefile ushare-1.1a/src/Makefile
--- ushare-1.1a-orig/src/Makefile	2007-12-09 13:03:36.000000000 +0000
+++ ushare-1.1a/src/Makefile	2015-05-06 15:20:14.466687537 +0100
@@ -53,7 +53,7 @@
 all: depend $(PROG)
 
 .c.o:
-	$(CC) -c $(CFLAGS) $(OPTFLAGS) -o $@ $<
+	$(CC) -c $(CFLAGS) $(OPTFLAGS) -I.. -o $@ $<
 
 $(PROG): $(OBJS)
 	$(CC) $(OBJS) $(LDFLAGS) $(EXTRALIBS) -o $@
diff -ru ushare-1.1a-orig/src/trace.c ushare-1.1a/src/trace.c
--- ushare-1.1a-orig/src/trace.c	2007-12-09 13:03:36.000000000 +0000
+++ ushare-1.1a/src/trace.c	2015-05-07 16:14:32.137822051 +0100
@@ -43,7 +43,7 @@
     return;
 
   va_start (va, format);
-  if (is_daemon)
+  if (1 || is_daemon)
   {
     int flags = LOG_DAEMON;
     flags |= level == ULOG_ERROR ? LOG_ERR : LOG_NOTICE;
diff -ru ushare-1.1a-orig/src/ushare.c ushare-1.1a/src/ushare.c
--- ushare-1.1a-orig/src/ushare.c	2007-12-09 13:03:36.000000000 +0000
+++ ushare-1.1a/src/ushare.c	2015-05-07 16:13:45.145451532 +0100
@@ -793,7 +793,7 @@
     ut->starting_id = STARTING_ENTRY_ID_XBOX360;
   }
 
-  if (ut->daemon)
+  if (1 || ut->daemon)
   {
     /* starting syslog feature as soon as possible */
     start_log ();
