--- linux-2.6.32.1/arch/arm/mach-orion5x/dns323-setup.c	2009-12-03 03:51:21.000000000 +0000
+++ linux-2.6.32.1-safe/arch/arm/mach-orion5x/dns323-setup.c	2009-12-17 18:03:04.174058287 +0000
@@ -32,6 +32,7 @@
 
 #define DNS323_GPIO_LED_RIGHT_AMBER	1
 #define DNS323_GPIO_LED_LEFT_AMBER	2
+#define DNS323_GPIO_SYSTEM_RUNNING  3
 #define DNS323_GPIO_LED_POWER		5
 #define DNS323_GPIO_OVERTEMP		6
 #define DNS323_GPIO_RTC			7
@@ -334,8 +335,8 @@
 	{  0, MPP_UNUSED },
 	{  1, MPP_GPIO },		/* right amber LED (sata ch0) */
 	{  2, MPP_GPIO },		/* left amber LED (sata ch1) */
-	{  3, MPP_UNUSED },
-	{  4, MPP_GPIO },		/* power button LED */
+	{  3, MPP_GPIO },		/* jc: power button LED */
+	{  4, MPP_UNUSED },		
 	{  5, MPP_GPIO },		/* power button LED */
 	{  6, MPP_GPIO },		/* GMT G751-2f overtemp */
 	{  7, MPP_GPIO },		/* M41T80 nIRQ/OUT/SQW */
@@ -377,6 +378,13 @@
 {
 	pr_info("%s: triggering power-off...\n", __func__);
 	gpio_set_value(DNS323_GPIO_POWER_OFF, 1);
+	/* jc: Based on Erik Benada patch
+	http://lists.arm.linux.org.uk/lurker/message/20090415.023343.ac77419a.en.html */
+	if (dns323_dev_id() == MV88F5182_DEV_ID) { 
+		set_current_state(TASK_UNINTERRUPTIBLE);
+		schedule_timeout(100);
+		gpio_set_value(DNS323_GPIO_POWER_OFF, 0);
+	}
 }
 
 static void __init dns323_init(void)
@@ -401,7 +409,11 @@
 	platform_device_register(&dns323_nor_flash);
 
 	platform_device_register(&dns323_gpio_leds);
-
+	if (dns323_dev_id() == MV88F5182_DEV_ID)
+	/* jc: Experimentally verified that on the 5182 gpio 3 must be set
+	to stop blinking (and start using) the power led */
+		gpio_set_value(DNS323_GPIO_SYSTEM_RUNNING, 1);
+		
 	platform_device_register(&dns323_button_device);
 
 	i2c_register_board_info(0, dns323_i2c_devices,
@@ -414,6 +426,10 @@
 		printk("DNS323: Failed to read MAC address\n");
 
 	orion5x_ehci0_init();
+	/* jc: The 5182 has a 2nd internal USB host used by some hardware people
+	if (dns323_dev_id() == MV88F5182_DEV_ID)
+		orion5x_ehci1_init();
+	*/
 	orion5x_eth_init(&dns323_eth_data);
 	orion5x_i2c_init();
 	orion5x_uart0_init();
