--- kexec/arch/arm/kexec-zImage-arm.c.orig	2009-06-07 21:34:01.000000000 +0100
+++ kexec/arch/arm/kexec-zImage-arm.c	2009-06-07 21:34:11.000000000 +0100
@@ -11,7 +11,7 @@
 #include <stdint.h>
 #include <getopt.h>
 #include <arch/options.h>
-#include <asm/page.h>
+//jc: #include <asm/page.h>
 #include "../../kexec.h"
 
 #define COMMAND_LINE_SIZE 1024
--- kexec/phys_arch.c.orig	2009-06-10 19:59:21.000000000 +0100
+++ kexec/phys_arch.c	2009-06-10 19:59:59.000000000 +0100
@@ -12,7 +12,7 @@
 			strerror(errno));
 		return -1;
 	}
-
+return arches[0].arch; // jc:
 	for (i = 0; arches[i].machine; ++i)
 		if (strcmp(utsname.machine, arches[i].machine) == 0)
 			return arches[i].arch;
