diff -ur configure configure
--- configure	2007-12-09 13:03:36.000000000 +0000
+++ configure	2009-09-14 20:07:25.000000000 +0100
@@ -248,7 +248,7 @@
 }
 
 check_lib_version() {
-    check_cmd pkg-config "$1" --atleast-version="$2"
+    check_cmd $PKG_CONFIG "$1" --atleast-version="$2"
     err=$?
     return $err
 }
@@ -434,10 +434,14 @@
   ;;
   --help) show_help
   ;;
+  --target=*) cross_compile="yes" # jc:
+  ;;
+  --host=*) cpu=${optval%-linux} # jc:
+  ;;
   *)
   echo "Unknown option \"$opt\"."
   echo "See $0 --help for available options."
-  exit 1
+	# jc: exit 1
   ;;
   esac
 done
@@ -646,8 +650,8 @@
 echolog "Checking for libupnp >= $libupnp_min_version ..."
 check_lib upnp/upnp.h UpnpSetMaxContentLength -lupnp || die "Error, can't find libupnp !"
 check_lib_version libupnp $libupnp_min_version || die "Error, libupnp < $libupnp_min_version !"
-add_cflags `pkg-config libupnp --cflags`
-add_extralibs `pkg-config libupnp --libs`
+add_cflags `$PKG_CONFIG libupnp --cflags`
+add_extralibs `$PKG_CONFIG libupnp --libs`
 
 #################################################
 #   check for libdlna (mandatory if enabled)
@@ -662,8 +666,8 @@
   check_lib dlna.h dlna_register_all_media_profiles -ldlna || die "Error, can't find libdlna (install it or use --disable-dlna) !"
   check_lib_version libdlna $libdlna_min_version || die "Error, libdlna < $libdlna_min_version !"
   add_cflags -DHAVE_DLNA
-  add_cflags `pkg-config libdlna --cflags`
-  add_extralibs `pkg-config libdlna --libs`
+  add_cflags `$PKG_CONFIG libdlna --cflags`
+  add_extralibs `$PKG_CONFIG libdlna --libs`
 fi
 
 #################################################
@@ -672,8 +676,8 @@
 echolog ""
 echolog "uShare: configure is OK"
 echolog "  version            $VERSION"
-echolog "  using libupnp      `pkg-config libupnp --modversion`"
-test $dlna = yes && echolog "  using libdlna      `pkg-config libdlna --modversion`"
+echolog "  using libupnp      `$PKG_CONFIG libupnp --modversion`"
+test $dlna = yes && echolog "  using libdlna      `$PKG_CONFIG libdlna --modversion`"
 echolog "configuration:"
 echolog "  install prefix     $PREFIX"
 echolog "  configuration dir  $sysconfdir"
diff -ur Makefile Makefile
--- Makefile	2007-12-09 13:03:36.000000000 +0000
+++ Makefile	2009-09-11 05:41:44.000000000 +0100
@@ -36,6 +36,7 @@
 	-$(RM) -f config.mak
 	-$(RM) -f config.h
 
+install-strip:  install
 
 install:
 	for subdir in $(SUBDIRS); do \ 
diff -ur src/Makefile src/Makefile
--- src/Makefile	2007-12-09 13:03:36.000000000 +0000
+++ src/Makefile	2009-09-14 20:22:01.000000000 +0100
@@ -53,7 +53,7 @@
 all: depend $(PROG)
 
 .c.o:
-	$(CC) -c $(CFLAGS) $(OPTFLAGS) -o $@ $<
+	$(CC) -I.. -c $(CFLAGS) $(OPTFLAGS) -o $@ $<
 
 $(PROG): $(OBJS)
 	$(CC) $(OBJS) $(LDFLAGS) $(EXTRALIBS) -o $@
