--- mediatomb-0.11.0/configure.ac-orig	2008-03-01 22:48:54.000000000 +0000
+++ mediatomb-0.11.0/configure.ac	2010-04-15 01:36:05.034404124 +0100
@@ -2894,7 +2894,8 @@
 dnl information.
     CXXFLAGS="$CXXFLAGS $INOTIFY_CXXFLAGS"
     AC_MSG_CHECKING([whether sys/inotify.h works])
-    AC_RUN_IFELSE(
+dnl jc:    AC_RUN_IFELSE(
+		AC_COMPILE_IFELSE(
         AC_LANG_PROGRAM([[#include <sys/inotify.h>]],
                          [[return (-1 == inotify_init());]]
                        ),
@@ -2904,7 +2905,8 @@
                 [
                     AC_MSG_RESULT([no, using own inotify headers])
                     AC_MSG_CHECKING([whether inotify-nosys.h works on this system])
-                    AC_RUN_IFELSE(
+dnl jc:    AC_RUN_IFELSE(
+						AC_COMPILE_IFELSE(
                         AC_LANG_PROGRAM([[#include "src/inotify-nosys.h"]],
                          [[return (-1 == inotify_init());]]
                         ),
--- mediatomb-0.11.0/src/url.cc-orig	2010-04-15 01:42:15.897406689 +0100
+++ mediatomb-0.11.0/src/url.cc	2010-04-15 01:42:58.552532121 +0100
@@ -75,7 +75,8 @@
 
     if (only_header)
     {
-        curl_easy_setopt(curl_handle, CURLOPT_NOBODY);
+//jc:        curl_easy_setopt(curl_handle, CURLOPT_NOBODY);
+        curl_easy_setopt(curl_handle, CURLOPT_NOBODY, 0); // jc: not sure
         curl_easy_setopt(curl_handle, CURLOPT_HEADERFUNCTION, URL::dl);
         curl_easy_setopt(curl_handle, CURLOPT_HEADERDATA, 
                          (void *)buffer.getPtr());
