--- mxml-2.6/Makefile.in-orig	2011-02-12 23:37:53.291241341 +0000
+++ mxml-2.6/Makefile.in	2011-02-12 23:49:44.141367113 +0000
@@ -52,7 +52,7 @@
 libdir		=	@libdir@
 mandir		=	@mandir@
 docdir		=	@docdir@
-BUILDROOT	=	$(DSTROOT)
+BUILDROOT	=	$(DESTDIR)
 
 
 #
@@ -88,7 +88,8 @@
 			mxml-node.o mxml-search.o mxml-set.o
 LIBOBJS		=	$(PUBLIBOBJS) mxml-private.o mxml-string.o
 OBJS		=	mxmldoc.o testmxml.o $(LIBOBJS)
-TARGETS		=	$(LIBMXML) mxmldoc testmxml mxml.xml doc/mxml.man
+#jc: TARGETS		=	$(LIBMXML) mxmldoc testmxml mxml.xml doc/mxml.man
+TARGETS		=	$(LIBMXML)
 
 
 #
@@ -133,21 +134,24 @@
 # Install everything...
 #
 
-install:	$(TARGETS) install-$(LIBMXML) install-libmxml.a
+install-strip: install
+
+# jc: install:	$(TARGETS) install-$(LIBMXML) install-libmxml.a
+install:	$(TARGETS) install-$(LIBMXML)
 	$(INSTALL_DIR) $(BUILDROOT)$(bindir)
-	$(INSTALL_BIN) mxmldoc $(BUILDROOT)$(bindir) 
-	$(INSTALL_DIR) $(BUILDROOT)$(docdir)
-	for file in $(DOCFILES); do \
-		$(INSTALL_MAN) $$file $(BUILDROOT)$(docdir); \
-	done
+	# jc: $(INSTALL_BIN) mxmldoc $(BUILDROOT)$(bindir) 
+	# jc: $(INSTALL_DIR) $(BUILDROOT)$(docdir)
+	# jc: for file in $(DOCFILES); do \
+	# jc: 	$(INSTALL_MAN) $$file $(BUILDROOT)$(docdir); \
+	# jc: done
 	$(INSTALL_DIR) $(BUILDROOT)$(includedir)
 	$(INSTALL_DATA) mxml.h $(BUILDROOT)$(includedir)
 	$(INSTALL_DIR) $(BUILDROOT)$(libdir)/pkgconfig
 	$(INSTALL_DATA) mxml.pc $(BUILDROOT)$(libdir)/pkgconfig
-	$(INSTALL_DIR) $(BUILDROOT)$(mandir)/man1
-	$(INSTALL_MAN) doc/mxmldoc.man $(BUILDROOT)$(mandir)/man1/mxmldoc.1
-	$(INSTALL_DIR) $(BUILDROOT)$(mandir)/man3
-	$(INSTALL_MAN) doc/mxml.man $(BUILDROOT)$(mandir)/man3/mxml.3
+	# jc: $(INSTALL_DIR) $(BUILDROOT)$(mandir)/man1
+	# jc: $(INSTALL_MAN) doc/mxmldoc.man $(BUILDROOT)$(mandir)/man1/mxmldoc.1
+	# jc: $(INSTALL_DIR) $(BUILDROOT)$(mandir)/man3
+	# jc: $(INSTALL_MAN) doc/mxml.man $(BUILDROOT)$(mandir)/man3/mxml.3
 
 install-libmxml.a:
 	$(INSTALL_DIR) $(BUILDROOT)$(libdir)
