--- e2fsprogs-1.41.14/lib/ext2fs/icount.c-orig	2017-01-22 19:20:48.977343688 +0000
+++ e2fsprogs-1.41.14/lib/ext2fs/icount.c	2017-01-22 19:23:39.689744124 +0000
@@ -339,9 +339,7 @@
 static struct ext2_icount_el *get_icount_el(ext2_icount_t icount,
 					    ext2_ino_t ino, int create)
 {
-	float	range;
 	int	low, high, mid;
-	ext2_ino_t	lowval, highval;
 
 	if (!icount || !icount->list)
 		return 0;
@@ -363,31 +361,7 @@
 	low = 0;
 	high = (int) icount->count-1;
 	while (low <= high) {
-#if 0
-		mid = (low+high)/2;
-#else
-		if (low == high)
-			mid = low;
-		else {
-			/* Interpolate for efficiency */
-			lowval = icount->list[low].ino;
-			highval = icount->list[high].ino;
-
-			if (ino < lowval)
-				range = 0;
-			else if (ino > highval)
-				range = 1;
-			else {
-				range = ((float) (ino - lowval)) /
-					(highval - lowval);
-				if (range > 0.9)
-					range = 0.9;
-				if (range < 0.1)
-					range = 0.1;
-			}
-			mid = low + ((int) (range * (high-low)));
-		}
-#endif
+		mid = ((unsigned)low + (unsigned)high) >> 1;
 		if (ino == icount->list[mid].ino) {
 			icount->cursor = mid+1;
 			return &icount->list[mid];
