--- linux-2.6.35.14/net/core/filter.c.orig	2011-08-16 15:48:22.000000000 +0100
+++ linux-2.6.35.14/net/core/filter.c	2011-08-16 16:16:43.000000000 +0100
@@ -292,9 +292,11 @@
 			mem[f_k] = X;
 			continue;
 		default:
+#ifdef CONFIG_BUG
 			WARN_RATELIMIT(1, "Unknown code:%u jt:%u tf:%u k:%u\n",
 				       fentry->code, fentry->jt,
 				       fentry->jf, fentry->k);
+#endif
 			return 0;
 		}
 
