--- linux-2.6.34-orig/arch/arm/mach-orion5x/dns323-setup.c	2010-05-16 22:17:36.000000000 +0100
+++ linux-2.6.34/arch/arm/mach-orion5x/dns323-setup.c	2010-08-14 01:41:44.000000000 +0100
@@ -277,7 +277,6 @@
 static struct gpio_led_platform_data dns323_led_data = {
 	.num_leds	= ARRAY_SIZE(dns323_leds),
 	.leds		= dns323_leds,
-	.gpio_blink_set = dns323_gpio_blink_set,
 };
 
 static struct platform_device dns323_gpio_leds = {
