diff -ruN minidlna-230410//configure minidlna-230410-new/configure
--- minidlna-230410//configure	1970-01-01 01:00:00.000000000 +0100
+++ minidlna-230410-new/configure	2010-04-28 15:43:54.017240662 +0100
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+echo >> Makefile
+echo "CC = $CC_FOR_TARGET" >> Makefile
+echo "CFLAGS =  $CFLAGS -I ${STAGING_DIR}/usr/include/ffmpeg" >> Makefile
+echo "STRIP = $STRIP" >> Makefile
+echo "STAGING_DIR = $STAGING_DIR" >> Makefile
+
+true
diff -ruN minidlna-230410//configure~ minidlna-230410-new/configure~
--- minidlna-230410//configure~	1970-01-01 01:00:00.000000000 +0100
+++ minidlna-230410-new/configure~	2010-04-28 15:43:53.985365830 +0100
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+echo >> Makefile
+echo "CC = $CC_FOR_TARGET" >> Makefile
+echo "CFLAGS =  $CFLAGS -I ${STAGING_DIR}/usr/include/ffmpeg" >> Makefile
+echo "STRIP = $STRIP" >> Makefile
+echo "STAGING_DIR = $STAGING_DIR" >> Makefile
+
+true
diff -ruN minidlna-230410//genconfig.sh minidlna-230410-new/genconfig.sh
--- minidlna-230410//genconfig.sh	2009-11-19 03:22:35.000000000 +0000
+++ minidlna-230410-new/genconfig.sh	2010-04-28 15:17:01.440365214 +0100
@@ -24,22 +24,22 @@
 
 # Detect if there are missing headers
 # NOTE: This check only works with a normal distro
-[ ! -e "/usr/include/sqlite3.h" ] && MISSING="libsqlite3 $MISSING"
-[ ! -e "/usr/include/jpeglib.h" ] && MISSING="libjpeg $MISSING"
-[ ! -e "/usr/include/libexif/exif-loader.h" ] && MISSING="libexif $MISSING"
-[ ! -e "/usr/include/id3tag.h" ] && MISSING="libid3tag $MISSING"
-[ ! -e "/usr/include/ogg/ogg.h" ] && MISSING="libogg $MISSING"
-[ ! -e "/usr/include/vorbis/codec.h" ] && MISSING="libvorbis $MISSING"
-[ ! -e "/usr/include/FLAC/metadata.h" ] && MISSING="libflac $MISSING"
-[ ! -e "/usr/include/ffmpeg/avutil.h" -a \
-  ! -e "/usr/include/libavutil/avutil.h" -a \
-  ! -e "/usr/include/ffmpeg/libavutil/avutil.h" ] && MISSING="libavutil $MISSING"
-[ ! -e "/usr/include/ffmpeg/avformat.h" -a \
-  ! -e "/usr/include/libavformat/avformat.h" -a \
-  ! -e "/usr/include/ffmpeg/libavformat/avformat.h" ] && MISSING="libavformat $MISSING"
-[ ! -e "/usr/include/ffmpeg/avcodec.h" -a \
-  ! -e "/usr/include/libavcodec/avcodec.h" -a \
-  ! -e "/usr/include/ffmpeg/libavcodec/avcodec.h" ] && MISSING="libavcodec $MISSING"
+[ ! -e "$STAGING_DIR/usr/include/sqlite3.h" ] && MISSING="libsqlite3 $MISSING"
+[ ! -e "$STAGING_DIR/usr/include/jpeglib.h" ] && MISSING="libjpeg $MISSING"
+[ ! -e "$STAGING_DIR/usr/include/libexif/exif-loader.h" ] && MISSING="libexif $MISSING"
+[ ! -e "$STAGING_DIR/usr/include/id3tag.h" ] && MISSING="libid3tag $MISSING"
+[ ! -e "$STAGING_DIR/usr/include/ogg/ogg.h" ] && MISSING="libogg $MISSING"
+[ ! -e "$STAGING_DIR/usr/include/vorbis/codec.h" ] && MISSING="libvorbis $MISSING"
+[ ! -e "$STAGING_DIR/usr/include/FLAC/metadata.h" ] && MISSING="libflac $MISSING"
+[ ! -e "$STAGING_DIR/usr/include/ffmpeg/avutil.h" -a \
+  ! -e "$STAGING_DIR/usr/include/libavutil/avutil.h" -a \
+  ! -e "$STAGING_DIR/usr/include/ffmpeg/libavutil/avutil.h" ] && MISSING="libavutil $MISSING"
+[ ! -e "$STAGING_DIR/usr/include/ffmpeg/avformat.h" -a \
+  ! -e "$STAGING_DIR/usr/include/libavformat/avformat.h" -a \
+  ! -e "$STAGING_DIR/usr/include/ffmpeg/libavformat/avformat.h" ] && MISSING="libavformat $MISSING"
+[ ! -e "$STAGING_DIR/usr/include/ffmpeg/avcodec.h" -a \
+  ! -e "$STAGING_DIR/usr/include/libavcodec/avcodec.h" -a \
+  ! -e "$STAGING_DIR/usr/include/ffmpeg/libavcodec/avcodec.h" ] && MISSING="libavcodec $MISSING"
 if [ -n "$MISSING" ]; then
 	echo -e "\nERROR!  Cannot continue."
 	echo -e "The following required libraries are either missing, or are missing development headers:\n"
diff -ruN minidlna-230410//inotify.c minidlna-230410-new/inotify.c
--- minidlna-230410//inotify.c	2010-01-13 21:15:10.000000000 +0000
+++ minidlna-230410-new/inotify.c	2010-04-28 05:32:12.410726974 +0100
@@ -52,6 +52,8 @@
 
 #define PATH_BUF_SIZE PATH_MAX
 
+#define rindex(a,b) strrchr((a),(b))
+
 struct watch
 {
 	int wd;		/* watch descriptor */
diff -ruN minidlna-230410//Makefile minidlna-230410-new/Makefile
--- minidlna-230410//Makefile	2010-03-18 21:37:39.000000000 +0000
+++ minidlna-230410-new/Makefile	2010-04-28 15:32:32.403240407 +0100
@@ -12,12 +12,12 @@
 #
 #CFLAGS = -Wall -O -D_GNU_SOURCE -g -DDEBUG
 #CFLAGS = -Wall -g -Os -D_GNU_SOURCE
-CFLAGS = -Wall -g -O3 -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 \
-	 -I/usr/include/ffmpeg \
-	 -I/usr/include/libavutil -I/usr/include/libavcodec -I/usr/include/libavformat \
-	 -I/usr/include/ffmpeg/libavutil -I/usr/include/ffmpeg/libavcodec -I/usr/include/ffmpeg/libavformat
+#CFLAGS = -Wall -g -O3 -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 \
+#	 -I/usr/include/ffmpeg \
+#	 -I/usr/include/libavutil -I/usr/include/libavcodec -I/usr/include/libavformat \
+#	 -I/usr/include/ffmpeg/libavutil -I/usr/include/ffmpeg/libavcodec -I/usr/include/ffmpeg/libavformat
 #STATIC_LINKING: LDFLAGS = -static
-CC = gcc
+#CC = gcc
 RM = rm -f
 INSTALL = install
 
@@ -53,12 +53,15 @@
 	$(RM) testupnpdescgen.o
 
 install:	minidlna
-	$(INSTALL) -d $(SBININSTALLDIR)
-	$(INSTALL) minidlna $(SBININSTALLDIR)
-	$(INSTALL) -d $(ETCINSTALLDIR)
-	$(INSTALL) --mode=0644 minidlna.conf $(ETCINSTALLDIR)
+	$(INSTALL) -d $(DESTDIR)/$(SBININSTALLDIR)
+	$(INSTALL) minidlna $(DESTDIR)/$(SBININSTALLDIR)
+	$(INSTALL) -d $(DESTDIR)/$(ETCINSTALLDIR)
+	$(INSTALL) --mode=0644 minidlna.conf $(DESTDIR)/$(ETCINSTALLDIR)
 
-minidlna:	$(BASEOBJS) $(LNXOBJS) $(LIBS)
+install-strip: install
+	$(STRIP) -s $(DESTDIR)/$(SBININSTALLDIR)/minidlna
+
+minidlna:	$(BASEOBJS) $(LNXOBJS)
 	@echo Linking $@
 	@$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(BASEOBJS) $(LNXOBJS) $(LIBS)
 
@@ -68,7 +71,7 @@
 	@$(CC) $(CFLAGS) -o $@ $(TESTUPNPDESCGENOBJS)
 
 config.h:	genconfig.sh
-	./genconfig.sh
+	STAGING_DIR=$(STAGING_DIR) ./genconfig.sh
 
 depend:	config.h
 	makedepend -f$(MAKEFILE_LIST) -Y \
@@ -122,8 +125,11 @@
 .SUFFIXES: .c .o
 
 .c.o:
-	@echo Compiling $*.c
-	@$(CC) $(CFLAGS) -o $@ -c $< && exit 0;\
-		echo "The following command failed:" 1>&2;\
-		echo "$(CC) $(CFLAGS) -o $@ -c $<";\
-		$(CC) $(CFLAGS) -o $@ -c $< &>/dev/null
+	$(CC) $(CFLAGS) -o $@ -c $<
+
+#.c.o:
+#	@echo Compiling $*.c
+#	@$(CC) $(CFLAGS) -o $@ -c $< && exit 0;\
+#		echo "The following command failed:" 1>&2;\
+#		echo "$(CC) $(CFLAGS) -o $@ -c $<";\
+#		$(CC) $(CFLAGS) -o $@ -c $< &>/dev/null
\ No newline at end of file
diff -ruN minidlna-230410//minidlna.c minidlna-230410-new/minidlna.c
--- minidlna-230410//minidlna.c	2009-11-19 03:22:35.000000000 +0000
+++ minidlna-230410-new/minidlna.c	2010-04-28 05:32:12.415600986 +0100
@@ -14,6 +14,7 @@
 #include <stdio.h>
 #include <ctype.h>
 #include <sys/types.h>
+#include <sys/stat.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
@@ -55,7 +56,9 @@
 # warning "Your SQLite3 library appears to be too old!  Please use 3.5.1 or newer."
 # define sqlite3_threadsafe() 0
 #endif
- 
+
+#define index(a,b) strchr((a),(b))
+
 /* OpenAndConfHTTPSocket() :
  * setup the socket used to handle incoming HTTP connections. */
 static int
diff -ruN minidlna-230410//scanner.c minidlna-230410-new/scanner.c
--- minidlna-230410//scanner.c	2010-01-13 21:15:11.000000000 +0000
+++ minidlna-230410-new/scanner.c	2010-04-28 05:32:12.416601261 +0100
@@ -36,6 +36,9 @@
 #include "scanner.h"
 #include "log.h"
 
+#define index(a,b) strchr((a),(b))
+#define rindex(a,b) strrchr((a),(b))
+
 int valid_cache = 0;
 
 struct virtual_item
diff -ruN minidlna-230410//tagutils/tagutils-misc.c minidlna-230410-new/tagutils/tagutils-misc.c
--- minidlna-230410//tagutils/tagutils-misc.c	2009-10-28 07:44:44.000000000 +0000
+++ minidlna-230410-new/tagutils/tagutils-misc.c	2010-04-28 05:32:12.417600931 +0100
@@ -26,6 +26,8 @@
 
 #define MAX_ICONV_BUF 1024
 
+#define index(a,b) strchr((a),(b))
+
 typedef enum {
 	ICONV_OK,
 	ICONV_TRYNEXT,
diff -ruN minidlna-230410//upnphttp.c minidlna-230410-new/upnphttp.c
--- minidlna-230410//upnphttp.c	2009-11-12 01:54:45.000000000 +0000
+++ minidlna-230410-new/upnphttp.c	2010-04-28 05:32:12.419601967 +0100
@@ -46,6 +46,8 @@
 //#define MAX_BUFFER_SIZE 4194304 // 4MB -- Too much?
 #define MAX_BUFFER_SIZE 2147483647 // 2GB -- Too much?
 
+#define index(a,b) strchr((a),(b))
+
 #include "icons.c"
 
 struct upnphttp * 
