*** Python-2.7.2/setup.py-orig	2012-05-03 19:50:56.329354224 +0100
--- Python-2.7.2/setup.py	2012-05-03 19:55:30.584983753 +0100
***************
*** 21,27 ****
  COMPILED_WITH_PYDEBUG = hasattr(sys, 'gettotalrefcount')
  
  # This global variable is used to hold the list of modules to be disabled.
! disabled_module_list = []
  
  def add_dir_to_list(dirlist, dir):
      """Add the directory 'dir' to the list 'dirlist' (at the front) if
--- 21,30 ----
  COMPILED_WITH_PYDEBUG = hasattr(sys, 'gettotalrefcount')
  
  # This global variable is used to hold the list of modules to be disabled.
! try:
!     disabled_module_list = os.environ["PYTHON_DISABLE_MODULES"].split() 
! except KeyError:
!     disabled_module_list = [] 
  
  def add_dir_to_list(dirlist, dir):
      """Add the directory 'dir' to the list 'dirlist' (at the front) if
***************
*** 382,387 ****
--- 385,393 ----
              add_dir_to_list(self.compiler.library_dirs, '/usr/local/lib')
              add_dir_to_list(self.compiler.include_dirs, '/usr/local/include')
              self.add_multiarch_paths()
+         else:
+             add_dir_to_list(self.compiler.library_dirs, '/home/jcard/Alt-F/build-pkgs/build_arm/staging_dir/usr/lib') 
+             add_dir_to_list(self.compiler.include_dirs, '/home/jcard/Alt-F/build-pkgs/build_arm/staging_dir/usr/include') 
  
          # Add paths specified in the environment variables LDFLAGS and
          # CPPFLAGS for header and library files.
