diff -ru busybox-1.20.2/libbb/correct_password.c busybox-1.20.2/libbb/correct_password.c
--- busybox-1.20.2/libbb/correct_password.c	2012-06-26 14:35:45.000000000 +0100
+++ busybox-1.20.2/libbb/correct_password.c	2018-12-30 17:39:52.330089294 +0000
@@ -30,6 +30,9 @@
 
 #include "libbb.h"
 
+#define PWD_BUFFER_SIZE 1024
+#define GRP_BUFFER_SIZE 1024
+
 /* Ask the user for a password.
  * Return 1 if the user gives the correct password for entry PW,
  * 0 if not.  Return 1 without asking if PW has an empty password.
@@ -44,7 +47,7 @@
 #if ENABLE_FEATURE_SHADOWPASSWDS
 	/* Using _r function to avoid pulling in static buffers */
 	struct spwd spw;
-	char buffer[256];
+	char buffer[PWD_BUFFER_SIZE];
 #endif
 
 	/* fake salt. crypt() can choke otherwise. */
diff -ru busybox-1.20.2/libpwdgrp/pwd_grp.c busybox-1.20.2/libpwdgrp/pwd_grp.c
--- busybox-1.20.2/libpwdgrp/pwd_grp.c	2012-06-26 14:35:45.000000000 +0100
+++ busybox-1.20.2/libpwdgrp/pwd_grp.c	2018-12-30 17:39:51.034114298 +0000
@@ -23,8 +23,8 @@
 /**********************************************************************/
 /* Sizes for statically allocated buffers. */
 
-#define PWD_BUFFER_SIZE 256
-#define GRP_BUFFER_SIZE 256
+#define PWD_BUFFER_SIZE 1024
+#define GRP_BUFFER_SIZE 1024
 
 /**********************************************************************/
 /* Prototypes for internal functions. */
diff -ru busybox-1.20.2/loginutils/passwd.c busybox-1.20.2/loginutils/passwd.c
--- busybox-1.20.2/loginutils/passwd.c	2012-06-26 14:35:45.000000000 +0100
+++ busybox-1.20.2/loginutils/passwd.c	2018-12-30 17:39:49.782138454 +0000
@@ -16,6 +16,9 @@
 #include "libbb.h"
 #include <syslog.h>
 
+#define PWD_BUFFER_SIZE 1024
+#define GRP_BUFFER_SIZE 1024
+
 static void nuke_str(char *str)
 {
 	if (str) memset(str, 0, strlen(str));
@@ -106,7 +109,7 @@
 #if ENABLE_FEATURE_SHADOWPASSWDS
 	/* Using _r function to avoid pulling in static buffers */
 	struct spwd spw;
-	char buffer[256];
+	char buffer[PWD_BUFFER_SIZE];
 #endif
 
 	logmode = LOGMODE_BOTH;
diff -ru busybox-1.20.2/loginutils/sulogin.c busybox-1.20.2/loginutils/sulogin.c
--- busybox-1.20.2/loginutils/sulogin.c	2012-06-26 14:35:45.000000000 +0100
+++ busybox-1.20.2/loginutils/sulogin.c	2018-12-30 17:39:48.442164313 +0000
@@ -14,6 +14,9 @@
 #include "libbb.h"
 #include <syslog.h>
 
+#define PWD_BUFFER_SIZE 1024
+#define GRP_BUFFER_SIZE 1024
+
 //static void catchalarm(int UNUSED_PARAM junk)
 //{
 //	exit(EXIT_FAILURE);
@@ -29,7 +32,7 @@
 	const char *shell;
 #if ENABLE_FEATURE_SHADOWPASSWDS
 	/* Using _r function to avoid pulling in static buffers */
-	char buffer[256];
+	char buffer[PWD_BUFFER_SIZE];
 	struct spwd spw;
 #endif
 
diff -ru busybox-1.20.2/networking/httpd.c busybox-1.20.2/networking/httpd.c
--- busybox-1.20.2/networking/httpd.c	2012-07-02 15:08:25.000000000 +0100
+++ busybox-1.20.2/networking/httpd.c	2018-12-30 17:39:45.250225891 +0000
@@ -125,6 +125,8 @@
 //usage:     "\n	-d STRING	URL decode STRING"
 
 #include "libbb.h"
+#define PWD_BUFFER_SIZE 1024
+#define GRP_BUFFER_SIZE 1024
 #if ENABLE_PAM
 /* PAM may include <locale.h>. We may need to undefine bbox's stub define: */
 # undef setlocale
@@ -1771,7 +1773,7 @@
 			char *colon_after_user;
 			const char *passwd;
 # if ENABLE_FEATURE_SHADOWPASSWDS && !ENABLE_PAM
-			char sp_buf[256];
+			char sp_buf[PWD_BUFFER_SIZE];
 # endif
 
 			colon_after_user = strchr(user_and_passwd, ':');
