--- forked-daapd-0.13/config.h.in-orig	2011-03-14 21:46:59.634888301 +0000
+++ forked-daapd-0.13/config.h.in	2011-03-14 21:48:45.257012263 +0000
@@ -1,5 +1,19 @@
 /* config.h.in.  Generated from configure.in by autoheader.  */
 
+/*  
+  uclibc does not has pipe2() neither inotify_init1(), use pipe() and inotify_init().
+  Might fail at runtime, because of the missing flags
+  The uclibc version in use also does not have fdtimer, a patch was applied, is it working?
+*/
+
+#define pipe2(a,b) pipe(a)
+#define inotify_init1(a) inotify_init()
+
+/* uclibc does not has htobexx, is this correct? */
+#define htobe32(a) htonl((a))
+#define htobe16(a) htons((a))
+#define be16toh(a) ntohs((a))
+
 /* Define to 1 if you have the <dirent.h> header file. */
 #undef HAVE_DIRENT_H
 
--- forked-daapd-0.13/src/raop.c-orig	2011-03-14 21:49:18.542011600 +0000
+++ forked-daapd-0.13/src/raop.c	2011-03-14 21:49:52.136888965 +0000
@@ -37,6 +37,7 @@
 
 #include <stdio.h>
 #include <unistd.h>
+#include <stdlib.h>
 #include <stdint.h>
 #include <inttypes.h>
 #include <math.h>
--- forked-daapd-0.13/src/laudio_alsa.c-orig	2011-03-14 21:50:12.390011203 +0000
+++ forked-daapd-0.13/src/laudio_alsa.c	2011-03-14 21:50:53.102011868 +0000
@@ -28,7 +28,7 @@
 #include <stdint.h>
 #include <inttypes.h>
 
-#include <asoundlib.h>
+#include <alsa/asoundlib.h>
 
 #include "conffile.h"
 #include "logger.h"
