--- forked-daapd-0.12/src/httpd_daap.c-orig	2011-02-22 06:11:31.757556609 +0000
+++ forked-daapd-0.12/src/httpd_daap.c	2011-02-22 06:14:30.878684347 +0000
@@ -1161,24 +1161,17 @@
   char *name;
   char *passwd;
   const char *clientver;
-  int supports_update;
   int mpro;
   int apro;
   int len;
   int ret;
 
-  /* We don't support updates atm */
-  supports_update = 0;
-
   lib = cfg_getsec(cfg, "library");
   passwd = cfg_getstr(lib, "password");
   name = cfg_getstr(lib, "name");
 
   len = 169 + strlen(name);
 
-  if (!supports_update)
-    len -= 9;
-
   ret = evbuffer_expand(evbuf, len);
   if (ret < 0)
     {
@@ -1226,8 +1219,8 @@
   dmap_add_char(evbuf, "mspi", 1);   /* 9 */
   dmap_add_int(evbuf, "msdc", 1);    /* 12 */
 
-  if (supports_update)
-    dmap_add_char(evbuf, "msup", 0); /* 9 */
+  /* Advertise updates support even though we don't send updates */
+  dmap_add_char(evbuf, "msup", 1);   /* 9 */
 
   httpd_send_reply(req, HTTP_OK, "OK", evbuf);
 }
