--- nfs-utils-1.2.5/support/include/sockaddr.h-orig	2011-12-17 19:01:08.484218572 +0000
+++ nfs-utils-1.2.5/support/include/sockaddr.h	2011-12-17 19:01:29.352778896 +0000
@@ -20,7 +20,6 @@
 #ifndef NFS_UTILS_SOCKADDR_H
 #define NFS_UTILS_SOCKADDR_H
 
-#include <libio.h>
 #include <stdbool.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
--- nfs-utils-1.2.5/support/nfs/conffile.c-orig	2011-12-17 19:03:10.094660532 +0000
+++ nfs-utils-1.2.5/support/nfs/conffile.c	2011-12-17 19:03:32.814183722 +0000
@@ -43,7 +43,6 @@
 #include <string.h>
 #include <unistd.h>
 #include <errno.h>
-#include <err.h>
 #include <syslog.h>
 
 #include "conffile.h"
--- nfs-utils-1.2.5/utils/statd/sm-notify.c-orig	2011-12-17 19:04:00.093611657 +0000
+++ nfs-utils-1.2.5/utils/statd/sm-notify.c	2011-12-17 19:04:10.109401733 +0000
@@ -8,7 +8,6 @@
 #include <config.h>
 #endif
 
-#include <err.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/stat.h>
