--- nfs-utils-1.2.4/configure.ac-orig	2011-09-19 19:38:20.466174266 +0100
+++ nfs-utils-1.2.4/configure.ac	2011-09-19 19:41:37.571154939 +0100
@@ -248,7 +248,7 @@
               [AC_MSG_ERROR([Function 'getservbyname' not found.])])
 
 AC_CHECK_LIB([crypt], [crypt], [LIBCRYPT="-lcrypt"])
-
+AM_CONDITIONAL(CONFIG_NFSIDMAP, [test "$enable_nfsidmap" = "yes"])
 if test "$enable_nfsv4" = yes; then
   dnl check for libevent libraries and headers
   AC_LIBEVENT
--- nfs-utils-1.2.4/support/include/sockaddr.h-orig	2011-09-19 19:49:28.521547347 +0100
+++ nfs-utils-1.2.4/support/include/sockaddr.h	2011-09-19 19:49:59.246288717 +0100
@@ -20,7 +20,7 @@
 #ifndef NFS_UTILS_SOCKADDR_H
 #define NFS_UTILS_SOCKADDR_H
 
-#include <libio.h>
+/* jc: #include <libio.h> */
 #include <stdbool.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
--- nfs-utils-1.2.4/support/nfs/conffile.c-orig	2011-09-19 19:51:09.174250653 +0100
+++ nfs-utils-1.2.4/support/nfs/conffile.c	2011-09-19 19:51:40.683035287 +0100
@@ -43,7 +43,7 @@
 #include <string.h>
 #include <unistd.h>
 #include <errno.h>
-#include <err.h>
+/* jc: #include <err.h> */
 #include <syslog.h>
 
 #include "conffile.h"
--- nfs-utils-1.2.4/utils/statd/sm-notify.c-orig	2011-09-19 21:10:16.346946333 +0100
+++ nfs-utils-1.2.4/utils/statd/sm-notify.c	2011-09-19 21:10:33.047887663 +0100
@@ -8,7 +8,7 @@
 #include <config.h>
 #endif
 
-#include <err.h>
+/* jc: #include <err.h> */
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/stat.h>
