--- nfs-utils-1.2.6/support/include/sockaddr.h.orig	2012-11-07 18:36:40.221932229 +0000
+++ nfs-utils-1.2.6/support/include/sockaddr.h	2012-11-07 18:41:23.630178024 +0000
@@ -20,7 +20,6 @@
 #ifndef NFS_UTILS_SOCKADDR_H
 #define NFS_UTILS_SOCKADDR_H
 
-#include <libio.h>
 #include <stdbool.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
--- nfs-utils-1.2.6/support/nfs/conffile.c.orig	2012-11-07 18:36:55.886614246 +0000
+++ nfs-utils-1.2.6/support/nfs/conffile.c	2012-11-07 18:41:23.630178024 +0000
@@ -43,7 +43,6 @@
 #include <string.h>
 #include <unistd.h>
 #include <errno.h>
-#include <err.h>
 #include <syslog.h>
 
 #include "conffile.h"
--- nfs-utils-1.2.6/utils/statd/sm-notify.c.orig	2012-11-07 18:37:09.325341438 +0000
+++ nfs-utils-1.2.6/utils/statd/sm-notify.c	2012-11-07 18:41:23.630178024 +0000
@@ -8,7 +8,6 @@
 #include <config.h>
 #endif
 
-#include <err.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/stat.h>
--- nfs-utils-1.2.6/utils/Makefile.in.orig	2012-11-07 18:37:51.468485899 +0000
+++ nfs-utils-1.2.6/utils/Makefile.in	2012-11-07 18:38:02.477262399 +0000
@@ -38,7 +38,7 @@
 @CONFIG_NFSV41_TRUE@am__append_3 = blkmapd
 @CONFIG_GSS_TRUE@am__append_4 = gssd
 @CONFIG_MOUNT_TRUE@am__append_5 = mount
-@CONFIG_NFSDCLD_TRUE@am__append_6 = nfsdcld
+@CONFIG_NFSDCLD_TRUE@am__append_6 = 
 subdir = utils
 DIST_COMMON = $(srcdir)/Makefile.am $(srcdir)/Makefile.in
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
@@ -79,8 +79,8 @@
 	distdir
 ETAGS = etags
 CTAGS = ctags
-DIST_SUBDIRS = exportfs mountd nfsd nfsstat showmount statd osd_login \
-	idmapd nfsidmap blkmapd gssd mount nfsdcld
+DIST_SUBDIRS = exportfs mountd nfsd nfsstat showmount statd  \
+	idmapd nfsidmap blkmapd gssd mount 
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 am__relativize = \
   dir0=`pwd`; \
@@ -288,7 +288,7 @@
 	nfsstat \
 	showmount \
 	statd \
-	osd_login \
+	 \
 	$(OPTDIRS)
 
 MAINTAINERCLEANFILES = Makefile.in
