--- support/nfs/nfs_mntent.c	2009-06-02 15:43:05.000000000 +0100
+++ support/nfs/nfs_mntent.c	2009-10-15 03:44:27.000000000 +0100
@@ -1,7 +1,7 @@
 /* Private version of the libc *mntent() routines. */
 /* Note slightly different prototypes. */
 
-/* 1999-02-22 Arkadiusz Mi¶kiewicz <misiek@pld.ORG.PL>
+/* 1999-02-22 Arkadiusz Miï¿½kiewicz <misiek@pld.ORG.PL>
  * - added Native Language Support
  *
  * 2006-06-08 Amit Gud <agud@redhat.com>
@@ -163,7 +163,8 @@
 			return NULL;
 
 		mfp->mntent_lineno++;
-		s = index (buf, '\n');
+		// jc: s = index (buf, '\n');
+		s = rawmemchr (buf, '\n');
 		if (s == NULL) {
 			/* Missing final newline?  Otherwise extremely */
 			/* long line - assume file was corrupted */
@@ -171,7 +172,8 @@
 				fprintf(stderr, _("[mntent]: warning: no final "
 					"newline at the end of %s\n"),
 					mfp->mntent_file);
-				s = index (buf, 0);
+				// jc: s = index (buf, 0);
+				s = rawmemchr (buf, 0);
 			} else {
 				mfp->mntent_errs = 1;
 				goto err;
--- utils/mount/error.c	2009-06-02 15:43:05.000000000 +0100
+++ utils/mount/error.c	2009-10-15 03:39:21.000000000 +0100
@@ -62,7 +62,8 @@
 	char *tmp;
 
 	if (estr) {
-		if ((ptr = index(estr, ':')))
+		// jc: if ((ptr = index(estr, ':')))
+		if ((ptr = rawmemchr(estr, ':')))
 			estr = ++ptr;
 
 		tmp = &errbuf[spos];
--- utils/mountd/fsloc.c	2009-06-02 15:43:05.000000000 +0100
+++ utils/mountd/fsloc.c	2009-10-15 03:37:12.000000000 +0100
@@ -126,7 +126,8 @@
 	struct servers *rv=NULL;
 
 	xlog(L_NOTICE, "method_list(%s)\n", data);
-	for (ptr--, listsize=1; ptr; ptr=index(ptr, ':'), listsize++)
+ 		// jc:	for (ptr--, listsize=1; ptr; ptr=index(ptr, ':'), listsize++)
+	for (ptr--, listsize=1; ptr; ptr=rawmemchr(ptr, ':'), listsize++)
 		ptr++;
 	list = malloc(listsize * sizeof(char *));
 	copy = strdup(data);
--- utils/mount/Makefile.in	2009-06-02 16:23:25.000000000 +0100
+++ utils/mount/Makefile.in	2009-10-15 04:15:26.000000000 +0100
@@ -14,6 +14,10 @@
 
 @SET_MAKE@
 
+# These binaries go in /sbin (not /usr/sbin), and that cannot be
+# overriden at config time.
+#jc: sbindir = /sbin
+
 VPATH = @srcdir@
 pkgdatadir = $(datadir)/@PACKAGE@
 pkglibdir = $(libdir)/@PACKAGE@
@@ -225,10 +229,7 @@
 prefix = @prefix@
 program_transform_name = @program_transform_name@
 psdir = @psdir@
-
-# These binaries go in /sbin (not /usr/sbin), and that cannot be
-# overriden at config time.
-sbindir = /sbin
+sbindir = @sbindir@
 sharedstatedir = @sharedstatedir@
 srcdir = @srcdir@
 startstatd = @startstatd@

