--- /home/jcard/tmp/nfs-utils-1.2.3/utils/statd/sm-notify.c	2010-09-28 13:24:16.000000000 +0100
+++ nfs-utils-1.2.3/utils/statd/sm-notify.c	2010-09-29 03:23:34.669043826 +0100
@@ -8,7 +8,7 @@
 #include <config.h>
 #endif
 
-#include <err.h>
+// jc: #include <err.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/stat.h>
--- /home/jcard/tmp/nfs-utils-1.2.3/support/nfs/conffile.c	2010-09-28 13:24:16.000000000 +0100
+++ nfs-utils-1.2.3/support/nfs/conffile.c	2010-09-29 03:22:18.177044973 +0100
@@ -43,7 +43,7 @@
 #include <string.h>
 #include <unistd.h>
 #include <errno.h>
-#include <err.h>
+// jc: #include <err.h>
 #include <syslog.h>
 
 #include "conffile.h"
--- /home/jcard/tmp/nfs-utils-1.2.3/support/include/sockaddr.h	2010-09-28 13:24:16.000000000 +0100
+++ nfs-utils-1.2.3/support/include/sockaddr.h	2010-09-29 03:24:49.643918431 +0100
@@ -20,7 +20,7 @@
 #ifndef NFS_UTILS_SOCKADDR_H
 #define NFS_UTILS_SOCKADDR_H
 
-#include <libio.h>
+// jc: #include <libio.h>
 #include <stdbool.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
