--- automatic-0.6.4/src/web.c-orig	2011-09-26 04:27:52.224607286 +0100
+++ automatic-0.6.4/src/web.c	2011-09-26 04:28:05.558339802 +0100
@@ -356,6 +356,7 @@
   curl_easy_setopt(curl, CURLOPT_FORBID_REUSE, 1L );
   curl_easy_setopt(curl, CURLOPT_MAXREDIRS, -1L );
   curl_easy_setopt(curl, CURLOPT_NOSIGNAL, 1L );
+  curl_easy_setopt(curl, CURLOPT_USERAGENT, "WhoCares");
 
   if(auth && *auth) {
     dbg_printf(P_INFO2, "auth: %s", auth);
