--- gs-8.71/base/lib.mak.orig	2009-12-29 22:18:01.000000000 +0000
+++ gs-8.71/base/lib.mak	2010-06-17 14:48:14.000000000 +0100
@@ -237,6 +237,13 @@
  $(memory__h) $(string__h) $(gp_h) $(gpgetenv_h) $(gpmisc_h)
 	$(GLCC) $(GLO_)gpmisc.$(OBJ) $(C_) $(GLSRC)gpmisc.c
 
+$(GLOBJ)gpmisc_host.$(OBJ) : $(GLSRC)gpmisc.c\
+ $(unistd__h) $(fcntl__h) $(stat__h) $(stdio__h)\
+ $(memory__h) $(string__h) $(gp_h) $(gpgetenv_h) $(gpmisc_h)
+	$(CCAUX) $(I_)$(GLSRCDIR)$(_I) $(I_)$(GLOBJDIR)$(_I) \
+ $(GLO_)gpmisc_host.$(OBJ) $(C_) $(GLSRC)gpmisc.c
+
+
 # Command line argument list management
 $(GLOBJ)gsargs.$(OBJ) : $(GLSRC)gsargs.c\
  $(ctype__h) $(stdio__h) $(string__h)\
@@ -250,10 +257,22 @@
  $(gpcheck_h) $(gserror_h) $(gxfarith_h) $(gxfixed_h) $(stdint__h) $(stdio__h)
 	$(GLCC) $(GLO_)gsmisc.$(OBJ) $(C_) $(GLSRC)gsmisc.c
 
+$(GLOBJ)gsmisc_host.$(OBJ) : $(GLSRC)gsmisc.c $(GXERR)\
+  $(gconfigv_h) $(std_h) $(vmsmath_h)\
+  $(ctype__h) $(malloc__h) $(math__h) $(memory__h) $(string__h)\
+  $(gpcheck_h) $(gserror_h) $(gxfarith_h) $(gxfixed_h) $(stdint__h)
+	$(CCAUX) $(I_)$(GLSRCDIR)$(_I) $(I_)$(GLOBJDIR)$(_I) \
+  $(GLO_)gsmisc_host.$(OBJ) $(C_) $(GLSRC)gsmisc.c
+
 $(GLOBJ)gslibctx.$(OBJ) : $(GLSRC)gslibctx.c  $(GXERR)\
   $(gslibctx_h) $(stdio__h)
 	$(GLCC) $(GLO_)gslibctx.$(OBJ) $(C_) $(GLSRC)gslibctx.c
 
+$(GLOBJ)gslibctx_host.$(OBJ) : $(GLSRC)gslibctx.c  $(GXERR)\
+   $(gslibctx_h) $(stdio__h)
+	$(CCAUX) $(I_)$(GLSRCDIR)$(_I) $(I_)$(GLOBJDIR)$(_I)\
+  $(GLO_)gslibctx_host.$(OBJ) $(C_) $(GLSRC)gslibctx.c
+
 $(GLOBJ)gsnotify.$(OBJ) : $(GLSRC)gsnotify.c $(GXERR)\
  $(gsnotify_h) $(gsstruct_h)
 	$(GLCC) $(GLO_)gsnotify.$(OBJ) $(C_) $(GLSRC)gsnotify.c
@@ -267,6 +286,12 @@
  $(gsrect_h) $(gsuid_h) $(gsutil_h) $(gzstate_h) $(gxdcolor_h)
 	$(GLCC) $(GLO_)gsutil.$(OBJ) $(C_) $(GLSRC)gsutil.c
 
+$(GLOBJ)gsutil_host.$(OBJ) : $(GLSRC)gsutil.c $(AK) $(memory__h) $(string__h)\
+ $(gstypes_h) $(gconfigv_h) $(gserror_h) $(gserrors_h) $(gsmemory_h)\
+ $(gsrect_h) $(gsuid_h) $(gsutil_h) $(gzstate_h) $(gxdcolor_h)
+	$(CCAUX) $(I_)$(GLSRCDIR)$(_I) $(I_)$(GLOBJDIR)$(_I)\
+ $(GLO_)gsutil_host.$(OBJ) $(C_) $(GLSRC)gsutil.c
+
 # MD5 digest
 md5_h=$(GLSRC)md5.h
 # We have to use a slightly different compilation approach in order to
@@ -538,6 +563,13 @@
 	$(CP_) $(GLSRC)gscdef.c $(GLGEN)gscdefs.c
 	$(GLCC) $(GLO_)gscdefs.$(OBJ) $(C_) $(GLGEN)gscdefs.c
 
+$(GLOBJ)gscdefs_host.$(OBJ) : $(GLSRC)gscdef.c\
+ $(std_h) $(gscdefs_h) $(gconfigd_h) $(TOP_MAKEFILES)
+	$(RM_) $(GLGEN)gscdefs.c
+	$(CP_) $(GLSRC)gscdef.c $(GLGEN)gscdefs.c
+	$(CCAUX) $(I_)$(GLSRCDIR)$(_I) $(I_)$(GLOBJDIR)$(_I)\
+ $(GLO_)gscdefs_host.$(OBJ) $(C_) $(GLGEN)gscdefs.c
+
 $(GLOBJ)gxacpath.$(OBJ) : $(GLSRC)gxacpath.c $(GXERR)\
  $(gsdcolor_h) $(gsrop_h) $(gsstate_h) $(gsstruct_h) $(gsutil_h)\
  $(gxdevice_h) $(gxfixed_h) $(gxistate_h) $(gxpaint_h)\
@@ -2715,6 +2747,11 @@
 MKROMFS_ZLIB_OBJS=$(GLOBJ)compress.$(OBJ) $(GLOBJ)deflate.$(OBJ) \
 	$(GLOBJ)zutil.$(OBJ) $(GLOBJ)adler32.$(OBJ) $(GLOBJ)crc32.$(OBJ) \
 	$(GLOBJ)trees.$(OBJ)
+
+MKROMFS_ZLIB_HOST_OBJS=$(GLOBJ)compress_host.$(OBJ) $(GLOBJ)deflate_host.$(OBJ) \
+  $(GLOBJ)zutil_host.$(OBJ) $(GLOBJ)adler32_host.$(OBJ) $(GLOBJ)crc32_host.$(OBJ) \
+  $(GLOBJ)trees_host.$(OBJ)
+
 MKROMFS_COMMON_DEPS=$(stdpre_h) $(stdint__h) $(gsiorom_h) $(arch_h)\
 	$(gsmemret_h) $(gsmalloc_h) $(gsstype_h) $(gp_h) $(time__h)
 
@@ -2781,6 +2818,12 @@
 $(GLOBJ)gp_upapr.$(OBJ) : $(GLSRC)gp_upapr.c $(AK) $(gp_h)
 	$(GLCC) $(GLO_)gp_upapr.$(OBJ) $(C_) $(GLSRC)gp_upapr.c
 
+$(GLOBJ)gp_getnv_host.$(OBJ) : $(GLSRC)gp_getnv.c $(AK) $(stdio__h) $(string__h)\
+ $(gp_h) $(gsmemory_h) $(gstypes_h)
+	$(CCAUX) $(I_)$(GLSRCDIR)$(_I) $(I_)$(GLOBJDIR)$(_I)\
+ $(GLO_)gp_getnv_host.$(OBJ) $(C_) $(GLSRC)gp_getnv.c
+
+
 # File system implementation.
 
 # MS-DOS file system, also used by Desqview/X.
@@ -2800,10 +2843,21 @@
  $(stat__h) $(dirent__h)
 	$(GLCC) $(GLO_)gp_unifs.$(OBJ) $(C_) $(GLSRC)gp_unifs.c
 
+$(GLOBJ)gp_unifs_host.$(OBJ) : $(GLSRC)gp_unifs.c $(AK)\
+  $(memory__h) $(string__h) $(stdio__h) $(unistd__h) \
+  $(gx_h) $(gp_h) $(gpmisc_h) $(gsstruct_h) $(gsutil_h) \
+  $(stat__h) $(dirent__h)
+	$(CCAUX) $(I_)$(GLSRCDIR)$(_I) $(I_)$(GLOBJDIR)$(_I)\
+  $(GLO_)gp_unifs_host.$(OBJ) $(C_) $(GLSRC)gp_unifs.c
+
 # Unix(-like) file name syntax, *not* used by Desqview/X.
 $(GLOBJ)gp_unifn.$(OBJ) : $(GLSRC)gp_unifn.c $(AK) $(gx_h) $(gp_h) $(gpmisc_h)
 	$(GLCC) $(GLO_)gp_unifn.$(OBJ) $(C_) $(GLSRC)gp_unifn.c
 
+$(GLOBJ)gp_unifn_host.$(OBJ) : $(GLSRC)gp_unifn.c $(AK) $(gx_h) $(gp_h) $(gpmisc_h)
+	$(CCAUX) $(I_)$(GLSRCDIR)$(_I) $(I_)$(GLOBJDIR)$(_I)\
+  $(GLO_)gp_unifn_host.$(OBJ) $(C_) $(GLSRC)gp_unifn.c
+
 # Pipes.  These are actually the same on all platforms that have them.
 
 pipe_=$(GLOBJ)gdevpipe.$(OBJ)
--- gs-8.71/base/unix-aux.mak.orig	2009-01-08 09:17:18.000000000 +0000
+++ gs-8.71/base/unix-aux.mak	2010-06-17 14:50:54.000000000 +0100
@@ -37,6 +37,13 @@
  $(gx_h) $(gsexit_h) $(gp_h)
 	$(GLCC) $(FONTCONFIG_CFLAGS) $(GLO_)gp_unix.$(OBJ) $(C_) $(GLSRC)gp_unix.c
 
+$(GLOBJ)gp_unix_host.$(OBJ): $(GLSRC)gp_unix.c $(AK)\
+ $(pipe__h) $(string__h) $(time__h)\
+ $(gx_h) $(gsexit_h) $(gp_h)
+	$(CCAUX) $(I_)$(GLSRCDIR)$(_I) $(I_)$(GLOBJDIR)$(_I)\
+ $(GLO_)gp_unix_host.$(OBJ) $(C_) $(GLSRC)gp_unix.c
+
+
 $(GLOBJ)gp_unix_cache.$(OBJ): $(GLSRC)gp_unix_cache.c $(AK)\
  $(stdio__h) $(string__h) $(time__h) $(gconfigd_h) $(gp_h) $(md5_h)
 	$(GLCC) $(GLO_)gp_unix_cache.$(OBJ) $(C_) $(GLSRC)gp_unix_cache.c
@@ -46,6 +53,12 @@
   $(stdio__h) $(time__h) $(unistd__h) $(gx_h) $(gp_h)
 	$(GLCC) $(GLO_)gp_stdia.$(OBJ) $(C_) $(GLSRC)gp_stdia.c
 
+$(GLOBJ)gp_stdia_host.$(OBJ): $(GLSRC)gp_stdia.c $(AK)\
+  $(stdio__h) $(time__h) $(unistd__h) $(gx_h) $(gp_h)
+	$(CCAUX) $(I_)$(GLSRCDIR)$(_I) $(I_)$(GLOBJDIR)$(_I)\
+ $(GLO_)gp_stdia_host.$(OBJ) $(C_) $(GLSRC)gp_stdia.c
+
+
 # System V platforms other than SVR4, which lack some system calls,
 # but have pipes.
 sysv__=$(GLOBJ)gp_getnv.$(OBJ) $(GLOBJ)gp_upapr.$(OBJ) $(GLOBJ)gp_unix.$(OBJ) $(GLOBJ)gp_unifs.$(OBJ) $(GLOBJ)gp_unifn.$(OBJ) $(GLOBJ)gp_sysv.$(OBJ)
@@ -77,13 +90,13 @@
 $(GENINIT_XE): $(GLSRC)geninit.c $(AK) $(GENINIT_DEPS)
 	$(CCAUX) $(I_)$(GLSRCDIR)$(_I) $(O_)$(GENINIT_XE) $(GLSRC)geninit.c
 
-MKROMFS_OBJS=$(MKROMFS_ZLIB_OBJS) $(GLOBJ)gscdefs.$(OBJ) $(GLOBJ)gsmisc.$(OBJ) \
- $(GLOBJ)gpmisc.$(OBJ) $(GLOBJ)gslibctx.$(OBJ) $(GLOBJ)gp_getnv.$(OBJ) \
- $(GLOBJ)gp_unix.$(OBJ) $(GLOBJ)gp_unifs.$(OBJ) $(GLOBJ)gp_unifn.$(OBJ) \
- $(GLOBJ)gp_stdia.$(OBJ) $(GLOBJ)gsutil.$(OBJ)
+MKROMFS_OBJS=$(MKROMFS_ZLIB_HOST_OBJS) $(GLOBJ)gscdefs_host.$(OBJ) $(GLOBJ)gsmisc_host.$(OBJ) \
+  $(GLOBJ)gpmisc_host.$(OBJ) $(GLOBJ)gslibctx_host.$(OBJ) $(GLOBJ)gp_getnv_host.$(OBJ) \
+  $(GLOBJ)gp_unix_host.$(OBJ) $(GLOBJ)gp_unifs_host.$(OBJ) $(GLOBJ)gp_unifn_host.$(OBJ) \
+  $(GLOBJ)gp_stdia_host.$(OBJ) $(GLOBJ)gsutil_host.$(OBJ)
 
 $(MKROMFS_XE): $(GLSRC)mkromfs.c $(MKROMFS_COMMON_DEPS) $(MKROMFS_OBJS)
-	$(CCAUX) $(GENOPT) $(CFLAGS) $(I_)$(GLSRCDIR)$(_I) $(I_)$(GLOBJ)$(_I) $(I_)$(ZSRCDIR)$(_I) $(GLSRC)mkromfs.c $(O_)$(MKROMFS_XE) $(MKROMFS_OBJS) -lm $(EXTRALIBS)
+	$(CCAUX) $(GENOPT) $(CFLAGS_DEBUG) $(I_)$(GLSRCDIR)$(_I) $(I_)$(GLOBJ)$(_I) $(I_)$(ZSRCDIR)$(_I) $(GLSRC)mkromfs.c $(O_)$(MKROMFS_XE) $(MKROMFS_OBJS) -lm
 
 # Query the environment to construct gconfig_.h.
 # The "else true;" is required because Ultrix's implementation of sh -e
--- gs-8.71/base/zlib.mak.orig	2007-06-05 23:23:38.000000000 +0100
+++ gs-8.71/base/zlib.mak	2010-06-17 14:50:06.000000000 +0100
@@ -74,6 +74,9 @@
 $(ZOBJ)zutil.$(OBJ) : $(ZSRC)zutil.c $(ZDEP)
 	$(ZCC) $(ZO_)zutil.$(OBJ) $(C_) $(ZSRC)zutil.c
 
+$(ZOBJ)zutil_host.$(OBJ) : $(ZSRC)zutil.c $(ZDEP)
+	$(CCAUX) $(ZO_)zutil_host.$(OBJ) $(C_) $(ZSRC)zutil.c
+
 # Encoding (compression) code.
 
 $(ZGEN)zlibe.dev : $(TOP_MAKEFILES) $(ZGEN)zlibe_$(SHARE_ZLIB).dev
@@ -91,16 +94,28 @@
 $(ZOBJ)adler32.$(OBJ) : $(ZSRC)adler32.c $(ZDEP)
 	$(ZCC) $(ZO_)adler32.$(OBJ) $(C_) $(ZSRC)adler32.c
 
+$(ZOBJ)adler32_host.$(OBJ) : $(ZSRC)adler32.c $(ZDEP)
+	$(CCAUX) $(ZO_)adler32_host.$(OBJ) $(C_) $(ZSRC)adler32.c
+
 $(ZOBJ)deflate.$(OBJ) : $(ZSRC)deflate.c $(ZDEP)
 	$(ZCC) $(ZO_)deflate.$(OBJ) $(C_) $(ZSRC)deflate.c
 
+$(ZOBJ)deflate_host.$(OBJ) : $(ZSRC)deflate.c $(ZDEP)
+	$(CCAUX) $(ZO_)deflate_host.$(OBJ) $(C_) $(ZSRC)deflate.c
+
 # new file in zlib 1.2.x
 $(ZOBJ)compress.$(OBJ) : $(ZSRC)compress.c $(ZDEP)
 	$(ZCC) $(ZO_)compress.$(OBJ) $(C_) $(ZSRC)compress.c
 
+$(ZOBJ)compress_host.$(OBJ) : $(ZSRC)compress.c $(ZDEP)
+	$(CCAUX) $(ZO_)compress_host.$(OBJ) $(C_) $(ZSRC)compress.c
+
 $(ZOBJ)trees.$(OBJ) : $(ZSRC)trees.c $(ZDEP)
 	$(ZCC) $(ZO_)trees.$(OBJ) $(C_) $(ZSRC)trees.c
 
+$(ZOBJ)trees_host.$(OBJ) : $(ZSRC)trees.c $(ZDEP)
+	$(CCAUX) $(ZO_)trees_host.$(OBJ) $(C_) $(ZSRC)trees.c
+
 # The zlib filters per se don't need crc32, but libpng versions starting
 # with 0.90 do.
 
@@ -118,6 +133,9 @@
 $(ZOBJ)crc32.$(OBJ) : $(ZSRC)crc32.c $(ZDEP)
 	$(CC_NO_WARN) $(ZCCFLAGS) $(ZO_)crc32.$(OBJ) $(C_) $(ZSRC)crc32.c
 
+$(ZOBJ)crc32_host.$(OBJ) : $(ZSRC)crc32.c $(ZDEP)
+	$(CCAUX) $(ZCCFLAGS) $(ZO_)crc32_host.$(OBJ) $(C_) $(ZSRC)crc32.c
+
 # Decoding (decompression) code.
 
 $(ZGEN)zlibd.dev : $(TOP_MAKEFILES) $(ZGEN)zlibd_$(SHARE_ZLIB).dev
--- gs-8.71/base/unixinst.mak.orig	2010-06-17 15:34:08.000000000 +0100
+++ gs-8.71/base/unixinst.mak	2010-06-16 04:35:19.000000000 +0100
@@ -15,6 +15,8 @@
 # containing the `install' targets.
 # This is the very last part of the makefile for these configurations.
 
+install-strip: install
+
 install: install-exec install-scripts install-data $(INSTALL_SHARED) $(INSTALL_CONTRIB)
 
 # The sh -c in the rules below is required because Ultrix's implementation
--- gs-8.71/cups/cups.mak.orig	2009-03-07 21:46:16.000000000 +0000
+++ gs-8.71/cups/cups.mak	2010-06-17 16:16:12.000000000 +0100
@@ -51,7 +51,7 @@
 
 $(PDFTORASTER_XE): $(pdftoraster_)
 	if [ "$(CUPSPDFTORASTER)" = "1" ]; then \
-	    $(GLCC) $(LDFLAGS) -DBINDIR='"$(bindir)"' -DGS='"$(GS)"' -o $@ $(pdftoraster_) `cups-config --image --libs`; \
+	    $(GLCC) $(LDFLAGS) -DBINDIR='"$(bindir)"' -DGS='"$(GS)"' -o $@ $(pdftoraster_) `$(CUPSCONFIG) --image --libs`; \
 	fi
 
 install:	install-cups
--- gs-8.71/base/configure.ac.orig	2009-11-24 08:36:27.000000000 +0000
+++ gs-8.71/base/configure.ac	2010-06-17 21:07:13.000000000 +0100
@@ -548,7 +548,7 @@
 AC_SUBST(CUPSSERVERROOT)
 AC_SUBST(CUPSDATA)
 AC_SUBST(CUPSPDFTORASTER)
-
+AC_SUBST(CUPSPCONFIG)
 
 dnl look for IJS implementation
 AC_ARG_WITH([ijs], AC_HELP_STRING([--without-ijs],
--- gs-8.71/base/Makefile.in.orig	2010-06-17 21:17:44.000000000 +0100
+++ gs-8.71/base/Makefile.in	2010-06-17 21:18:45.000000000 +0100
@@ -83,6 +83,7 @@
 CUPSSERVERROOT=@CUPSSERVERROOT@
 CUPSDATA=@CUPSDATA@
 CUPSPDFTORASTER=@CUPSPDFTORASTER@
+CUPSCONFIG=@CUPSCONFIG@
 
 # Define the default directory/ies for the runtime
 # initialization and font files.  Separate multiple directories with a :.
@@ -478,7 +479,7 @@
 
 CCFLAGS=$(GENOPT) $(CAPOPT) $(CFLAGS)
 CC_=$(CC) $(CCFLAGS)
-CCAUX=$(CC) $(CFLAGS)
+CCAUX=gcc -O0
 CC_LEAF=$(CC_)
 # note gcc can't use -fomit-frame-pointer with -pg.
 CC_LEAF_PG=$(CC_)
