--- libdlna-0.2.3/configure-orig	2010-04-25 17:21:52.000000000 +0100
+++ libdlna-0.2.3/configure	2010-04-25 17:51:18.000000000 +0100
@@ -427,7 +427,7 @@
   *)
   echo "Unknown option \"$opt\"."
   echo "See $0 --help for available options."
-  exit 1
+# exit 1
   ;;
   esac
 done
@@ -627,7 +627,8 @@
   return 0;
 }
 EOF
-VERSION=`$TMPE`
+#VERSION=`$TMPE`
+VERSION=0.2.4
 restore_flags
 
 
--- libdlna-0.2.3/Makefile-orig	2010-04-25 18:17:36.000000000 +0100
+++ libdlna-0.2.3/Makefile	2010-04-25 18:22:29.000000000 +0100
@@ -25,7 +25,7 @@
   LDFLAGS += $(EXTRALIBS)
 endif
 
-all: lib test
+all: lib
 
 lib:
 	$(MAKE) -C src
