--- linux-4.19.136/drivers/net/ethernet/marvell/mvmdio.c	2020-08-04 13:52:02.047492280 +0000
+++ linux-4.19.136/drivers/net/ethernet/marvell/mvmdio.c	2020-08-04 17:29:04.983422491 +0000
@@ -319,6 +319,7 @@
 
 	init_waitqueue_head(&dev->smi_busy_wait);
 
+#ifdef CONFIG_ARCH_ORION5X_DT
 	for (i = 0; i < ARRAY_SIZE(dev->clk); i++) {
 		dev->clk[i] = of_clk_get(pdev->dev.of_node, i);
 		if (PTR_ERR(dev->clk[i]) == -EPROBE_DEFER) {
@@ -329,6 +330,16 @@
 			break;
 		clk_prepare_enable(dev->clk[i]);
 	}
+#else
+	/* non-DT setup doesn't support multiple clocks using of_clk_get */
+	dev->clk[0] = devm_clk_get(&pdev->dev, NULL);
+	if (PTR_ERR(dev->clk[0]) == -EPROBE_DEFER) {
+		ret = -EPROBE_DEFER;
+		goto out_clk;
+	}
+	if (!IS_ERR(dev->clk[0]))
+		clk_prepare_enable(dev->clk[0]);
+#endif
 
 	dev->err_interrupt = platform_get_irq(pdev, 0);
 	if (dev->err_interrupt > 0 &&
